source/slang/slang-emit-metal.cpp

```diff
@@ -538,7 +538,8 @@ bool MetalSourceEmitter::tryEmitInstExprImpl(IRInst* inst, const EmitOpInfo& inO
 
 void MetalSourceEmitter::emitVectorTypeNameImpl(IRType* elementType, IRIntegerValue elementCount)
 {
-    emitSimpleTypeImpl(elementType);
+    // NM: Passing count here, as Metal 64-bit vector type names do not match their scalar equivalents.
+    emitSimpleTypeKnowingCount(elementType, elementCount);
```

Comment:
I think you can special case the 64bit types in the switch statement below, and emit `vec<T, N>` instead of `TN` for those types. We shouldn't need the `emitSimpleTypeKnowingCount()` change.